### PR TITLE
[Hotfix] Limit max gas used per trade

### PR DESCRIPTION
**Pull request by fleupold**
We are sometimes spending ridiculous amounts of gas on simple trades (e.g. https://etherscan.io/tx/0x2336894792550a59c2ac314c88a442aee3daed27b4aecc976943ebdcc9051a20)

This PR makes is so that we cap the amount of gas per trade at 750k

### Test Plan
No test, please review carefully.

### CODE REVIEW

**Review by marcovc — APPROVED** [review]
Looks good!

**Review by bh2smith — APPROVED** [review]

**Review by bh2smith — COMMENTED** [review]
Nice that you included metrics for this. 

Wait. I think I may have misread but it might be nice to capture these events in a metric.

File: solver/src/driver.rs [hunk 1]
```diff
@@ -411,6 +418,12 @@ fn liquidity_with_price(
     liquidity
 }
 
+// Returns true if the average gas cost in the settlement doesn't exceed MAX_GAS_PER_TRADE
```
Comment (bh2smith) [inline solver/src/driver.rs:421]:
```suggestion
// Returns true if the average gas cost (per trade) in the settlement doesn't exceed MAX_GAS_PER_TRADE
```

**Review by bh2smith — APPROVED** [review]
Just throwing some ideas around in the changes to ensure I have reviewed carefully. Everything looks good otherwise.

**Review by josojo — APPROVED** [review]

**Comment by fleupold** [comment]
Superseeded by #580

**Pull request closed without merging**